docs/adr/0001-add-generic-fetcher.md

```diff
@@ -0,0 +1,104 @@
+# Add generic fetcher
+
+## Context
+
+The main motivation for this change is to cover use cases of users that need to download arbitrary files that don't fit
```

Comment:
Non-blocking suggestion:
```diff
- The main motivation for this change is to cover use cases of users that need to download arbitrary files that don't fit
+ Some users need to download arbitrary files that don't fit
```

(since the document is not describing an individual change, but a new pseudo-package manager; also a bit easier to read)

Reply:
Fixed.